### DISCUSSION TITLE
Changing the metric used to calculate best.pt in YOLOv8

### DISCUSSION THREAD

**Discussion by SundarakrishnanN** [ultralytics/ultralytics#15639] [General]
1. Hey, How do we change the metric used to compute best.pt which by default is MAP-50 i believe. Is it possible to use something else like class loss or precision or recall? If so how do we do it in YOLOv8?

2. Secondly increasing the weight value of cls parameter increases its importance of predicting the correct class right?

**Comment by glenn-jocher** [comment]
@SundarakrishnanN to change the metric for computing `best.pt` in YOLOv8, you can modify the `save_best` parameter in the training configuration to your desired metric such as precision or recall. Regarding the `cls` parameter, increasing its weight indeed emphasizes the importance of predicting the correct class. For more details, please refer to the [Ultralytics documentation](https://docs.ultralytics.com).

**Comment by SundarakrishnanN** [comment]
Thank you so much for the support @glenn-jocher ! Also could you please share more details on how I could use the `save_best` parameter. I couldnt find about it in the documentation and neither could i find that as an argument in the  [Source code](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/engine/trainer.py) .

**Reply by glenn-jocher** [reply]
@SundarakrishnanN you're welcome! The `save_best` parameter isn't explicitly documented, but you can customize the metric for saving the best model by modifying the training loop in the source code. For detailed guidance, please refer to the [hyperparameter tuning guide](https://docs.ultralytics.com/guides/hyperparameter-tuning/).
